client/src/Components/Navbar.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Link, useNavigate, useParams } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {faGlobe} from '@fortawesome/free-solid-svg-icons'
+import i18 from 'i18next'
+import { useAuth0 } from "@auth0/auth0-react";
+
+
+const Navbar = () => {
+  const [dropdownVisible, setDropdownVisible] = useState(false);
+  const {t} =  useTranslation();
+  const {lng} = useParams(); //capture the language from url
+  const navigate = useNavigate();
+  const { user, isLoading, isAuthenticated, loginWithRedirect, logout } = useAuth0();
+
+  const {ul1,ul2,ul3,ul4,ul5} = t('ulTranslation')
+  const {login,register,logOut} = t('logRegTranslation')
+  const {Technology,Music,Literature,Politics,Culture} = t('blogTopics')
+
+  useEffect(()=>{
+    const savedLang = localStorage.getItem('i18nextLng') || 'en';
+    i18.changeLanguage(savedLang);
+  },[])
+  
+  const handleLanguageChange = (lang) => {
+    const currentPath = window.location.pathname;
+    const newPath = currentPath.replace(`/${i18.language}`, `/${lang}`);
+    i18.changeLanguage(lang);
+    localStorage.setItem('i18nextLng', lang);
+    navigate(newPath);
+    setDropdownVisible(false);
+  };
+  
+  // const getLanguagePath = () => `/${i18.language}`
+
+  if(isLoading)
+  {
+    return <div>Loading...</div>
+  }
+
+  return (
+    <div className='gap-[1150px]'>
+
+      {/* This is the 1st navbar */}
+      <div className= "bg-green-700}">
+        <div className={`flex bg-green-700 gap-[1150px] ${i18.language === 'ur' ? 'gap-[1130px]' : 'gap-[1150px'}
+                                                          ${i18.language === 'de' ? 'gap-[1090px]' : 'gap-[1140px'}`}>
+          <div className="flex flex-row items-center">
+            <Link to={`/${i18.language}`}>
+              <h3 className='text-white mr-3 ml-4'>DeUrdu</h3>
+            </Link>
+            <Link to={`/${i18.language}`}>
+              <h3 className='text-white'> ڈی اردو </h3>
+            </Link>
+          </div>
+            <div className="flex gap-1">
+               {/* Globe button with onClick to toggle dropdown */}
+              <button 
+                className='px-2 py-0.5 m-1 rounded-lg text-white relative'
```

Comment:
Are the styling/formatting parameters auto-generated, if not then how does the developer/test see the impact of small changes without relaunching the client side?

Reply:
Impact of small changes could be seen sudden, while using `tailwind` sudden and not leaving the root file.
While using raw CSS, it is much hard to handle the large code base.